Dockerfile

```diff
@@ -1,34 +1,60 @@
-FROM debian:stable-slim as fetcher
+# Copyright (c) 2018, 2019 Oracle and/or its affiliates. All rights reserved.
 
-# TODO copy just the 'distribution' directory, not all source code
+FROM ubuntu:bionic as build
+
+RUN apt-get update && apt-get install -y openjdk-8-jdk maven python3 python3-venv
+
+# Create a first layer to cache the "Maven World" in the local repository.
+# Incremental docker builds will always resume after that, unless you update the pom
+WORKDIR /mvn
+COPY pom.xml /mvn/
+COPY opengrok-indexer/pom.xml /mvn/opengrok-indexer/
+COPY opengrok-web/pom.xml /mvn/opengrok-web/
+COPY plugins/pom.xml /mvn/plugins/
+COPY suggester/pom.xml /mvn/suggester/
+
+# distribution and opengrok-tools do not have dependencies to cache
+RUN sed -i 's:<module>distribution</module>::g' /mvn/pom.xml
+RUN sed -i 's:<module>opengrok-tools</module>::g' /mvn/pom.xml
```

Comment:
why is this necessary ?

Reply:
Because if they are present, then `mvn` call will try to execute the plugins in those submodules which will fail the build. And I did not want to create other "dummy" files for it to pass with no added value.